pkg/schema/type.go

```diff
@@ -26,6 +26,8 @@ type Type interface {
 	SetTitle(string)
 	GetExamples() []Example
 	SetExamples([]Example)
+	GetDeprecated() bool
+	SetDeprecated(bool)
```

Comment:
Since this is our API if users are integrating with ytt programmatically, users may need to know two things:
1. _is_ the value deprecated
2. what the value for the deprecated string is (even though we do not yet use this, we still can expose this info. A integrator may want this info before we get around to prioritizing the story to use it, so lets make it available)

Lets change the Type interface to expose these things:
```
type Type interface {
...
   IsDeprecated() bool
   SetDeprecated(string)             <- takes the string instead of a bool
   GetDeprecationNotice() string
...
}
```

Reply:
After talking about this over zoom, we dicided to change the interface to:
```
IsDeprecated() (bool, string)
SetDeprecated(bool, string)
```

Reply:
Thanks Car! I have made the changes as discussed.